pkg/catalog/list.go

```diff
@@ -0,0 +1,49 @@
+package catalog
+
+import (
+	"fmt"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/weaveworks/pctl/pkg/subscription"
+)
+
+// ProfileData data containing profile and available version update for format printing.
+type ProfileData struct {
+	Profile                 subscription.SubscriptionSummary
+	AvailableVersionUpdates []string
+}
+
+// List will fetch all installed profiles on the cluster and check if there are updated versions available.
+func List(k8sClient runtimeclient.Client, catalogClient CatalogClient) ([]ProfileData, error) {
+	profiles, err := subscription.NewManager(k8sClient).List()
+	if err != nil {
+		return nil, err
+	}
+	if len(profiles) == 0 {
+		fmt.Println("no profiles found")
+		return nil, nil
+	}
+	profileData := make([]ProfileData, 0)
+	for _, p := range profiles {
+		var versions []string
+		// skip for profiles which don't have a catalog entry. i.e.: profiles installed via branch, url, path.
+		if p.Catalog != "-" {
+			related, err := GetAvailableUpdates(catalogClient, p.Catalog, p.Profile, p.Version)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get available updates: %w", err)
+			}
```

Comment:
do we want to error if we can't get the available updates? We could just put `-` or `n/a` and log an error, but still proceed? Not sure, maybe leave it like this and we can always come back to it? Talking myself out of it haha...

Reply:
This error is now definitive, like, if the output can't be marshalled. And later on things like, if the catalog doesn't respond. I think we want to error in those cases? The rest are ignored and `-` is used.

If we output something, we might break the formatter so I don't want to do that. That could mess with the table / json output.